lib/tool_shed/metadata/repository_metadata_manager.py

```diff
@@ -23,15 +35,200 @@
+        self.shed_config_dict = {}
+        self._reset_attributes_after_repository_update(relative_install_dir)
+
+    def handle_repository_elem(self, repository_elem, only_if_compiling_contained_td=False) -> HandleResultT:
+        """
+        Process the received repository_elem which is a <repository> tag either from a
+        repository_dependencies.xml file or a tool_dependencies.xml file.  If the former,
+        we're generating repository dependencies metadata for a repository in the Tool Shed.
+        If the latter, we're generating package dependency metadata within Galaxy or the
+        Tool Shed.
+        """
+        is_valid = True
+        error_message = ""
+        toolshed = repository_elem.get("toolshed", None)
+        name = repository_elem.get("name", None)
+        owner = repository_elem.get("owner", None)
+        changeset_revision = repository_elem.get("changeset_revision", None)
+        prior_installation_required = str(repository_elem.get("prior_installation_required", False))
+        repository_dependency_tup = [
+            toolshed,
+            name,
+            owner,
+            changeset_revision,
+            prior_installation_required,
+            str(only_if_compiling_contained_td),
+        ]
+        if not toolshed:
+            # Default to the current tool shed.
+            toolshed = self.trans.repositories_hostname
+            log.warning(f"\n\n\n\n\n\nin not toolshed with {toolshed}\n\n\n\n")
+            # toolshed = str(url_for("/", qualified=True)).rstrip("/")
+            repository_dependency_tup[0] = toolshed
+        else:
+            log.warning(f"moooocww.....{toolshed}\n\n\n\n\n")
+        toolshed = common_util.remove_protocol_from_tool_shed_url(toolshed)
+
+        if suc.tool_shed_is_this_tool_shed(toolshed, trans=self.trans):
+            try:
+                user = (
+                    self.sa_session.query(self.app.model.User)
+                    .filter(self.app.model.User.table.c.username == owner)
+                    .one()
+                )
+            except Exception:
+                error_message = (
+                    f"Ignoring repository dependency definition for tool shed {toolshed}, name {name}, owner {owner}, "
+                )
+                error_message += f"changeset revision {changeset_revision} because the owner is invalid."
+                log.debug(error_message)
+                is_valid = False
+                return repository_dependency_tup, is_valid, error_message
+            try:
+                repository = (
+                    self.sa_session.query(self.app.model.Repository)
+                    .filter(
+                        and_(
+                            self.app.model.Repository.table.c.name == name,
+                            self.app.model.Repository.table.c.user_id == user.id,
+                        )
+                    )
+                    .one()
+                )
```

Comment:
```suggestion
                repository = (
                    self.sa_session.execute(select(self.app.model.Repository)
                    .where(
                        and_(
                            self.app.model.Repository.name == name,
                            self.app.model.Repository.user_id == user.id,
                        )
                    )
                    ).scalar_one()
                )
```
formatting will be off here (my browser is barely handling this inline editing..)